make/data/autoheaders/assemblyprefix.h

```diff
@@ -0,0 +1,27 @@
+#
+# Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+# DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+#
+# This code is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 2 only, as
+# published by the Free Software Foundation.
+#
+# This code is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+# version 2 for more details (a copy is included in the LICENSE file that
+# accompanied this code).
+#
+# You should have received a copy of the GNU General Public License version
+# 2 along with this work; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+# or visit www.oracle.com if you need additional information or have any
+# questions.
+#
+
+// ASSEMBLY_SRC_FILE gets replaced by relative or absolute file path
+// in NativeCompilation.gmk, this ensures reproducible .debuginfo
```

Comment:
Actually, this ensures reproducible object file (`.o`) rather than the contents of `.debuginfo`. As far as I remember, `gcc` uses the "value" of `.file` in `.debuginfo`, but `clang` does not. Which is why the path mapping option is still necessary even with `.file` present and with a value stable across builds.

Maybe phrase this something like
```
... this ensures a reproducible object file through a predictable value of the STT_FILE symbol.
```
?

Reply:
@mkartashev thanks. So at the moment we are only using this assembly header for gcc tooling, see NativeCompliation.gmk change.
So maybe I change this to specifically state gcc and it's effect on linux debuginfo ?

Reply:
updated

Reply:
@andrew-m-leonard Thank you (both for making the change and correcting me wrt the scope of this commit, which is gcc-only).